### PR TITLE
#5406

### PR DESCRIPTION
**Pull request by reddevillg**

### CODE REVIEW

**Comment by Maijin** [comment]
If you add Fix #XXXX in the commit message it closes the issue automatically when merge.

**Comment by Maijin** [comment]
Also as you can see, this breaks two tests in the Travis CI i.e:

```
[33m[  ]  3  cmd_pf: [0mtypes                         
[31m[XX][0m

[35mRegression: [0mrun_tests.sh
[35mCommand: [0m /home/travis/build/radare/radare2/install/bin/radare2 -e scr.color=0 -N -q -i /tmp/r2-regressions//cmd_pf-EnW4Dj/rad   malloc://1024 > /tmp/r2-regressions//cmd_pf-EnW4Dj/out 2> /tmp/r2-regressions//cmd_pf-EnW4Dj/err
[35mFile: [0mmalloc://1024
[35mScript: [0mwx a00f00002a4b9a0200002a
pf fcbiC foo bar fool beer plop

[35mDiff: [0m--- /tmp/r2-regressions//cmd_pf-EnW4Dj/exp	2016-07-28 07:38:01.956869206 +0000
+++ /tmp/r2-regressions//cmd_pf-EnW4Dj/out	2016-07-28 07:38:01.968869442 +0000
@@ -1,4 +1,4 @@
- foo : 0x00000000 = 4000.000000
+ foo : 0x00000000 = 0.000000
  bar : 0x00000004 = '*'
 fool : 0x00000005 = 0x4b
 beer : 0x00000006 = 666
```

AND

```
[33m[  ]  16  cmd_pf: [0mfixed size variable           
[31m[XX][0m

[35mRegression: [0mrun_tests.sh
[35mCommand: [0m /home/travis/build/radare/radare2/install/bin/radare2 -e scr.color=0 -N -q -i /tmp/r2-regressions//cmd_pf-F5IJ9R/rad   malloc://1024 > /tmp/r2-regressions//cmd_pf-F5IJ9R/out 2> /tmp/r2-regressions//cmd_pf-F5IJ9R/err
[35mFile: [0mmalloc://1024
[35mScript: [0mwx a00f0000000000002a412fa00f000000000000402a44adde00000000efbeadde00000000000000000000000040
pf [2]i[3]b[2]f[3]c[3]w[2]q[6]x

[35mDiff: [0m--- /tmp/r2-regressions//cmd_pf-F5IJ9R/exp	2016-07-28 07:38:02.176873515 +0000
+++ /tmp/r2-regressions//cmd_pf-F5IJ9R/out	2016-07-28 07:38:02.188873750 +0000
@@ -1,6 +1,6 @@
 0x00000000 = [ 4000, 0 ]
 0x00000008 = [ 0x2a, 0x41, 0x2f ]
-0x0000000b = [ 4000.000000, 0.000000 ]
+0x0000000b = [ 0.000000, 0.000000 ]
 0x00000013 = [ '@', '*', 'D' ]
 0x00000016 = [ 0xdead, 0x0000, 0x0000 ]
 0x0000001c = (qword)[ 0x00000000deadbeef, 0x0000000000000000 ]
```

File: libr/util/p_format.c [hunk 1]
```diff
@@ -711,18 +711,18 @@ static void r_print_format_float(const RPrint* p, int endian, int mode,
...
 	} else if (mode & R_PRINT_DOT) {
 		//p->cb_printf ("%s", setval);
-		p->cb_printf ("%f", (float)addr);
+		p->cb_printf ("%f", *(float*)&addr);
 	} else {
 		if (MUSTSEE)
 			if (!SEEVALUE) p->cb_printf ("0x%08"PFMT64x" = ", seeki+((elem>=0)?elem*4:0));
 		if (size==-1)
-			p->cb_printf ("%f", (float)addr);
+			p->cb_printf ("%f", *(float*)&addr);
```
Comment (radare) [inline libr/util/p_format.c]: Not endian happy

File: libr/util/p_format.c [hunk 2]
```diff
@@ -711,18 +711,18 @@ static void r_print_format_float(const RPrint* p, int endian, int mode,
 		p->cb_printf ("wv4 %s @ 0x%08"PFMT64x"\n", setval, seeki+((elem>=0)?elem*4:0));
 	} else if (mode & R_PRINT_DOT) {
 		//p->cb_printf ("%s", setval);
-		p->cb_printf ("%f", (float)addr);
+		p->cb_printf ("%f", *(float*)&addr);
 	} else {
 		if (MUSTSEE)
 			if (!SEEVALUE) p->cb_printf ("0x%08"PFMT64x" = ", seeki+((elem>=0)?elem*4:0));
 		if (size==-1)
```
Comment (radare) [inline libr/util/p_format.c]: add braces even for 1 line conditionals

File: libr/util/p_format.c (hunk 1 above)
Comment (reddevillg) [inline libr/util/p_format.c]: i think `updateAddr` already handle it?

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
Fixed in master

**Pull request closed without merging**